lib/spack/spack/config.py

```diff
@@ -815,6 +836,8 @@ def create() -> Configuration:
     # No site-level configs should be checked into spack by default.
     configuration_paths.append(("site", os.path.join(spack.paths.etc_path)))
 
+    _add_configuration_paths_from_entry_points(configuration_paths)
```

Comment:
(minor) While here I would:
```suggestion
    configuration_paths.extend(paths_from_entry_points())
```